pkg/R/DataFrame.R

```diff
@@ -193,11 +363,62 @@ setMethod("take",
             collect(limited)
           })
 
+#' Head
+#'
+#' Return the first NUM rows of a DataFrame as a data.frame. If NUM is NULL, 
+#' then head() returns the first 6 rows in keeping with the current data.frame 
+#' convention in R.
+#'
+#' @param x A SparkSQL DataFrame
+#' @param num The number of rows to return. Default is 6.
+#' @return A data.frame
+#'
+#' @rdname head
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' head(df)
+#' }
+
+setMethod("head",
+          signature(x = "DataFrame"),
+          function(x, num = 6L) {
+          # Default num is 6L in keeping with R's data.frame convention
+            take(x, num)
+          })
+
+#' Return the first row of a DataFrame
+#'
+#' @param x A SparkSQL DataFrame
+#'
+#' @rdname first
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' first(df)
+#' }
+
+setGeneric("first", function(x) {standardGeneric("first") })
+
+setMethod("first",
+          signature(x = "DataFrame"),
+          function(x) {
+            take(x, 1)
```

Comment:
Hmmm, good question. Right now, `take` (and, by extension, `first`) call `collect` on the result, but `collect` returns a `data.frame` so it's not exactly a list, but it is a `data.frame` with only 1 row. It's not returning a formal `row` object like the standard API does, but returning a `data.frame` seemed like the most reasonable response in R since it gives you a clear and easily accessed object within R.